helper/locale.js

```diff
@@ -0,0 +1,94 @@
+module.exports = {
+
+  // returns the correct supported locale given the LANG variable
+  getLocale() {
+    const angularConf = require('../angular.json');
+    const i18n = angularConf.projects.OctoDash.i18n;
+    let lang;
+    try {
+      // Detect if environment LANG is set
+      // if it is defined, convert LANG from ie 'en_US.UTF-8' to 'en-US'
+      // if not defined, set it to the source locale
+      lang = process.env['LANG']
+        ? process.env['LANG'].split('.')[0].replace('_', '-')
+        : i18n.sourceLocale.code;
+    } catch(e) {
+      // LANG was populated with something else than a standard locale code
+      lang = i18n.sourceLocale.code;
+    }
+    // Check if exact LANG exists in our locales
+    const exactLocale = Object.keys(i18n.locales).includes(lang) && lang;
+    // Make the short version, convert ie 'en-US' to 'en'
+    const shortLang = lang.split('-')[0];
+    // Check if short LANG exists in our locales
+    // This matches 'fr-CA' to 'fr' if 'fr-CA' is not defined but 'fr' is
+    const approximateLocale = Object.keys(i18n.locales).includes(shortLang) && shortLang;
+    // Define locale by either the exact, approximate, or source locale in this order
+    const locale = exactLocale || approximateLocale || i18n.sourceLocale.code;
+    console.info('selected language: ' + locale);
```

Comment:
we could discuss this in a separate issue, i think the app lacks logging, and should default to a non-verbose logging scheme, but the verbosity should be increase-able to help the users document their issues

Reply:
Sure some logging probably is a good idea. The main problem here is on how these logs are shown to the users. Maybe there is a way to have a checkbox in the settings which enables verbose logging and also opens an HTTP server which will serve the logs, so they can be easily accessed or show them directly on the screen.

Asking the user to open dev tools / read some arbitrary file might not be the best user experience. I'll search npm and open a separate issue on this, would be great if you could remove this from this PR though, we can then add a logger and check what is possible there :)

Reply:
Ok created https://github.com/UnchartedBull/OctoDash/issues/1659. Let me know what you think, I'm really excited about this one and probably will start working on this once the onscreen keyboard is all buttoned-up.

Reply:
I propose keeping the console statement as i think a log statement should be put here, it's easy to change it to the next logging mechanism, does not eat memory or disk space. We can search and replace all console statements when ready. What do you think ? Can remove otherwise.

Reply:
Fine for the backend part, frontend part might cause issues as mentioned in the other issue :)